help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+        if args.defdir:
+            if not os.path.exists(args.defdir):
+                sys.exit("The defdir doesn't exist. Please try again.")
+            else:
+                for root_dir, dirs, files in os.walk(args.defdir):
+                    for filename in files:
+                        if os.path.splitext(filename)[1] == '.json':
+                            definition_files.append(os.path.join(root_dir, filename))
+
+        if definitions: survey_payload['definitions'] = definitions
+
+        # add sigmarule to list
+        if args.sigmarule:
+            sigma_rules.append(args.sigmarule)
+
+        # if --sigmadir, add all files to sigma_rules list
+        if args.sigmadir:
+            for root_dir, dirs, files in os.walk(args.sigmadir):
+                for filename in files:
+                    if os.path.splitext(filename)[1] == '.yml':
+                        sigma_rules.append(os.path.join(root_dir, filename))
+
+        if sigma_rules: survey_payload['sigma_rules'] = sigma_rules
+
+        # run search based on IOC file
+        if args.iocfile:
+            with open(args.iocfile) as ioc_file:
+                basename = os.path.basename(args.iocfile)
+                ioc_source = basename
+                ioc_list = [x.strip() for x in ioc_list]
+
+            survey_payload.update({"ioc_source": ioc_source, "ioc_list": ioc_list, "ioc_type": args.ioctype})
+
+        survey_payload['product_args'] = product_arg_builder(args=args)
+
+        survey_payload = {k:v for k,v in survey_payload.items() if v != None}
+        return survey_payload
+    
+def logger(edr:str, logs_dir:str='logs') -> logging.Logger:
+    # instantiate a logger
+    log = logging.getLogger('surveyor')
+    logging.debug(f'Product: {edr}')
+
+    # configure logging
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+    root.handlers = list()  # remove all default handlers
+    log_format = '[%(asctime)s] [%(levelname)-8s] [%(name)-36s] [%(filename)-20s:%(lineno)-4s] %(message)s'
+
+    # create logging directory if it does not exist
+    os.makedirs(logs_dir, exist_ok=True)
+
+    # create logging file handler
+    log_file_name = datetime.utcnow().strftime('%Y%m%d%H%M%S') + f'.{edr}.log'
+    handler = logging.FileHandler(os.path.join(logs_dir, log_file_name))
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(log_format))
+    root.addHandler(handler)
+    
+    return log
+
+def check_product_args_structure(edr: str, product_args: dict) -> dict:
+    if product_args == {}:
+        return {"result": True, "edr": edr, "required_fields": "No arguments were supplied"}
+
+    product_check = [k for k, v in product_args.items()]
+    
+    if edr == "cbc":
+        required_fields = ['device_group', 'device_policy']
+    elif edr == "cbr":
+        required_fields = ['sensor_group']
+    elif edr == "s1":
+        required_fields = ['deep_visibility']
+    else:
```

Comment:
DFE and Cortex do not take in product arguments at the moment. Cortex does have `_tenant_ids` but that's being account for in the creds dictionary, since it's an optional part of authentication for the EDR. Should I put in a placeholder or comment?

Reply:
Ah, that makes sense! Personally, I'd add a comment for if/when we need to make changes